File: DeepSpaceRuby/lib/PowerEfficientSpaceStation.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module Deepspace
require_relative 'Transformation'
require_relative 'PowerEfficientSpaceStationToUI'
require_relative 'SpaceStation'

  class PowerEfficientSpaceStation < SpaceStation
    @@EFFICIENCYFACTOR=1.10  
    
    def initialize(ss)
    super(ss.name,ss.ammoPower, ss.fuelUnits, ss.shieldPower, ss.weapons, ss.shieldBoosters,ss.hangar,ss.nMedals, ss.pendingDamage)
    end
    
    def getUIversion
      PowerEfficientSpaceStationToUI.new(self)
    end
    
    def fire
      super*@@EFFICIENCYFACTOR
    end
    
    def protection
      super*@@EFFICIENCYFACTOR
    end
    
    def setLoot(loot)
    super(loot)
    Transformation::NOTRANSFORM
  end
  
  end
end

File: DeepSpaceJava/src/deepspace/PowerEfficientSpaceStation.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package deepspace;

/**
 *
 * @author samuel
 */
public class PowerEfficientSpaceStation extends SpaceStation {
    static float EFFICIENCYFACTOR=(float) 1.10;
    
    public PowerEfficientSpaceStation(SpaceStation station){
        super (station);
    }
    
    public float fire(){
        return super.fire()*EFFICIENCYFACTOR;
    }
    
    public float protection(){
        return super.protection()*EFFICIENCYFACTOR;
    }
    
    public Transformation setLoot(Loot loot){
           super.setLoot(loot);
           return Transformation.NOTRANSFORM;
    }
    
    public PowerEfficientSpaceStationToUI getUIversion(){
        return new PowerEfficientSpaceStationToUI(this);
    }
}
